src/main/java/uk/gov/hmcts/reform/idam/web/AppController.java

```diff
@@ -382,6 +407,92 @@ private String authoriseUser(List<String> cookies, HttpServletRequest httpReques
         return responseUrl;
     }
 
+    private void initiateOtpFlow(AuthorizeRequest request,
+                                   List<String> cookies,
+                                   String ipAddress,
+                                   HttpServletResponse response,
+                                   Model model) {
+        final List<String> responseCookies = spiService.initiateOtpeAuthentication(cookies, ipAddress);
+        log.info("/login: Successful initiate OTP request - {}", obfuscateEmailAddress(request.getUsername()));
+        List<String> secureCookies = makeCookiesSecure(responseCookies);
+        secureCookies.forEach(cookie -> response.addHeader(HttpHeaders.SET_COOKIE, cookie));
+
+        model.addAttribute("authorizeCommand", request);
+    }
+
+    /**
+     * @should submit otp authentication using authId cookie and otp code then call authorise and redirect the user
+     * @should return verification view for INCORRECT_OTP 401 response
+     * @should return login view for non INCORRECT_OTP 401 response
+     * @should return login view for 403 response
+     * @should return login view when authorize fails
+     */
+    @PostMapping("/verification")
+    public ModelAndView verification(@ModelAttribute("authorizeCommand") AuthorizeRequest request,
```

Comment:
I know I named this method but I think it could be called `otpVerification()` to make it clearer.